### PR TITLE
ztp: PGT supports status updates or overlays

### PR DESCRIPTION
**Pull request by nishant-parekh**
Signed-off-by: Nishant Parekh <nparekh@redhat.com>

### CODE REVIEW

File: ztp/policygenerator/policyGen/policyBuilder.go [hunk 1]
```diff
@@ -246,6 +246,14 @@ func (pbuilder *PolicyBuilder) getCustomResource(sourceFile utils.SourceFile, so
 		resourceMap["data"] = make(map[string]interface{})
 		resourceMap["data"] = pbuilder.setValues(resourceMap["data"].(map[string]interface{}), sourceFile.Data)
 	}
+	if resourceMap["status"] != nil {
+		resourceMap["status"] = pbuilder.setValues(resourceMap["status"].(map[string]interface{}), sourceFile.Status)
+	} else if sourceFile.Status != nil {
+		// If the user supplies a "spec" section but the source CR does not have
```
Comment (serngawy) [inline ztp/policygenerator/policyGen/policyBuilder.go:252]: "status" not "spec"

**Review by serngawy — CHANGES_REQUESTED** [review]
can we keep in sync the CRD as well https://github.com/openshift-kni/cnf-features-deploy/blob/master/ztp/ran-crd/policy-gen-template-crd.yaml

File: ztp/policygenerator/policyGen/policyBuilder.go (hunk 1 above)
Comment (nishant-parekh) [reply ztp/policygenerator/policyGen/policyBuilder.go]: updated

**Comment by nishant-parekh** [comment]
> can we keep in sync the CRD as well https://github.com/openshift-kni/cnf-features-deploy/blob/master/ztp/ran-crd/policy-gen-template-crd.yaml

synced with this change.

File: ztp/ran-crd/policy-gen-template-crd.yaml [hunk 2]
```diff
@@ -179,3 +179,15 @@ spec:
...
+                          type: object
+                          description: |
+                            (optional) status contains the values that will be set in
+                            the source policy following the exact same path. If the
+                            status is not defined, the defined status in the
+                            sourcePolicies/{fileName} will be carried out to the
+                            generated custom resource. If any of the status' item
+                            fields its value starts with $, the item will be deleted
+                            from the generated data items if it is not setted in
+                            the sourceFile.status.
```
[the remark this answers is not in the document]
Comment (nishant-parekh) [reply ztp/ran-crd/policy-gen-template-crd.yaml]: removed that section from the other two sections as well.

**Review by imiller0 — COMMENTED** [review]
/lgtm

**Pull request merged**